docs/index.md

```diff
@@ -460,6 +460,35 @@ async def app(scope, receive, send):
     })
 ```
 
+## The WSGI Support
+
+Uvicorn offer support to WSGI applications throught `WSGIMiddleware` which converts
+your WSGI into ASGI application seamlessly just by running it with `--interface wsgi`
+
+!!! warning
+    The present WSGIMiddleware will be deprecated in favour of [a2wsgi](https://github.com/abersheeran/a2wsgi).
+    If you are using Uvicorn to run WSGI applications, please add it as part of your project requirements.
+
+To use the WSGI interface install its dependency:
+
+```shell
+pip install a2wsgi
+```
+
+Create an application, in `example.py`:
+
+```python
+def app(environ, start_response):
+    start_response('200 OK', [('Content-Type', 'text/plain')])
+    return [b'Hello World!']
+```
+
+Run the server:
+
+```shell
+$ uvicorn --interface wsgi example:app
```

Comment:
Is the `--interface` needed, or does uvicorn infers it? I never used WSGI with uvicorn. :thinking:

Reply:
It fails to infer it, maybe a bug but it falls into asgi2 case that also uses a function instead of a coroutine in the config code:
```python
...
            elif inspect.isfunction(self.loaded_app):
                use_asgi_3 = asyncio.iscoroutinefunction(self.loaded_app)
...
```